great_expectations/validator/validation_graph.py

```diff
@@ -53,6 +54,23 @@ def id(self) -> Tuple[str, str, str]:
             self.metric_value_kwargs_id,
         )
 
+    def to_json_dict(self) -> dict:
+        json_dict: dict = {
+            "metric_name": self.metric_name,
+            "metric_domain_kwargs": self.metric_domain_kwargs,
+            "metric_domain_kwargs_id": self.metric_domain_kwargs_id,
+            "metric_value_kwargs": self.metric_value_kwargs,
+            "metric_value_kwargs_id": self.metric_value_kwargs_id,
+            "id": self.id,
+        }
+        return json_dict
+
+    def __repr__(self):
+        return json.dumps(self.to_json_dict(), indent=2)
+
+    def __str__(self):
+        return self.__repr__()
+
```

Comment:
1. Could we please make `to_json_dict` a private method if it is only to be used with troubleshooting? A docstring or comment about its usage would further help.

Reply:
2. This is highly stylistic preference but I like to put dunder methods together. Could we move `__repr__` and `__str__` to be below `__init__`? Thoughts on that?

Reply:
3. Could we just use `self.__dict__()` and append missing attributes (specifically `ids`)? This I don't particularly feel strongly about I just tend to always be wary of listing out each individual attribute when defining `__repr__`.

Reply:
Doing it now...

Reply:
@cdkini I would like to keep the last as is -- because of a pattern we have elsewhere (e.g., `BatchRequest`, `Domain`, etc.) where we explicitly list the properties that matter for troubleshooting (most of the keys in `__dict__` do not, and even when there are exceptions, one can access them via the IDE, or print them out, etc. as needed.  Thanks.

Reply:
@alexsherstinsky that sounds very fair please do what works best for you then!

Reply:
@cdkini Actually, I have to revert one change: `to_json_dict` must be public, since we might promote `MetricConfiguration` to `MetricConfiguration(SerializableDotDict)` at some point.  Thanks.